src/diffusers/models/unet_2d_condition.py

```diff
@@ -127,20 +127,44 @@ def __init__(
         num_class_embeds: Optional[int] = None,
         upcast_attention: bool = False,
         resnet_time_scale_shift: str = "default",
+        time_embedding_type: str = "positional",  # fourier, positional
+        timestep_act_2: bool = False,
+        time_cond_proj_dim: Optional[int] = None,
+        conv_in_kernel: int = 3,
+        conv_out_kernel: int = 3,
+        attn1_types: Tuple[bool, None] = (None, "self", "self", "self"),
+        attn2_types: Tuple[bool, None] = (None, "cross", "cross", "cross"),
```

Comment:
This can be removed by deciding **inside** the KDown and KUp blocks what attention we need